frontend/src/components/Conversation/CodeBlock.tsx

```diff
@@ -65,40 +62,52 @@ function getNewCursorPosition(
   return i;
 }
 
-const formattedCodeOrInitial = (code: string) => {
+type SupportedFormatters =
+  | "bigquery"
+  | "db2"
+  | "hive"
+  | "mariadb"
+  | "mysql"
+  | "n1ql"
+  | "plsql"
+  | "postgresql"
+  | "redshift"
+  | "spark"
+  | "sql"
+  | "tsql";
+
+const formattedCodeOrInitial = (code: string, dialect: SupportedFormatters) => {
   try {
-    return format(code, { language: Dialect.Postgres });
+    return format(code, { language: dialect });
   } catch {
     return code;
   }
 };
 
 export const CodeBlock = ({
   code,
-  language,
+  dialect,
   resultId,
-  updateSQLRunResult,
-  updateChartResult,
+  onUpdateSQLRunResult,
+  onUpdateChartResult,
   forChart = false,
 }: {
   code: string;
   resultId: string;
-  language: IResultTypeName;
-  updateSQLRunResult: (sql_string_result_id: string, arg: string) => void;
-  updateChartResult: (
+  dialect: string;
+  onUpdateSQLRunResult: (sql_string_result_id: string, arg: string) => void;
+  onUpdateChartResult: (
     sql_string_result_id: string,
     newJson: string,
     newCreatedAt: string
   ) => void;
   forChart: boolean;
 }) => {
-  const { conversationId } = useParams({ from: "/_app/chat/$conversationId" });
-
   const [savedCode, setSavedCode] = useState<string>(() =>
-    formattedCodeOrInitial(code)
+    formattedCodeOrInitial(code, dialect as SupportedFormatters)
```

Comment:
What if the dialect is not supported? Like snowflake

Reply:
Not sure what would happen, gotta try it